src/api/test-controller/index.js

```diff
@@ -76,8 +76,8 @@ const originalThen = Promise.resolve().then;
 let inDebug = false;
 
 export default class TestController {
-    constructor (testRun) {
-        this._executionContext = null;
+    constructor (testRun, executionContext = null) {
+        this._executionContext = executionContext;
```

Comment:
To make code more compact, it's better to calculate the `_executionContext` property inside of `TestController` class:
 ```js
export default class TestController {
     constructor (testRun, executionContext) {
            this._executionContext = executionContext || testRun.controller?.getExecutionContext();
    }
}
```

Reply:
I removed these changes because they were mistaken and broke some tests.